### PR TITLE
pybind/ceph_argparse: avoid int overflow

### PR DESCRIPTION
**Pull request by tchaikov**
in python 2.6.8, `thread.join(timeout)` tries to convert the given
timeout to PyTime, but turns out `2 << 32` overflows when python
runtime converts the timeout from sec to ns. that's why
the `lock.acquire()` call always fail in `Thread._wait_for_tstate_lock()`.
and we end up with an alive thread after calling `thread.join()`.

Signed-off-by: Kefu Chai <kchai@redhat.com>

## Checklist
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard backend`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

File: src/pybind/ceph_argparse.py [hunk 1]
```diff
@@ -1316,7 +1316,7 @@ def run_in_thread(func, *args, **kwargs):
     if timeout == 0 or timeout == None:
         # python threading module will just get blocked if timeout is `None`,
         # otherwise it will keep polling until timeout or thread stops.
-        timeout = 2 ** 32
+        timeout = (1 << (32 - 1)) - 1
```
Comment (rzarzynski) [inline src/pybind/ceph_argparse.py]: Is there any difference in comparison to just `2**31 - 1`?
Is the `(32 - 1)` supposed to bring more attention on the integer's size in bits? If so, a comment could make this clearer.

Comment (tchaikov) [reply src/pybind/ceph_argparse.py]: no.
sure.

Comment (tchaikov) [reply src/pybind/ceph_argparse.py]: @rzarzynski  fixed and repushed. could you take another look?

**Comment by tchaikov** [comment]
http://pulpito.ceph.com/kchai-2020-02-08_06:31:22-rados-wip-kefu-testing-2020-02-08-1047-distro-basic-smithi/

**Pull request merged**